### PR TITLE
Updated alertmanager docs - Mails not sent when using remote smarthost with tls disabled

### PR DESCRIPTION
**Pull request by lyveng**
@brian-brazil As discussed in the [prometheus-users google group](https://groups.google.com/forum/#!topic/prometheus-users/_g-Z4e_Rya4), I've updated the docs. Please review.

### CODE REVIEW

File: content/docs/alerting/configuration.md [hunk 1]
```diff
@@ -78,7 +78,9 @@ global:
   [ smtp_auth_identity: <string> ]
   # SMTP Auth using CRAM-MD5. 
   [ smtp_auth_secret: <secret> ]
-  # The default SMTP TLS requirement.
+  # The default SMTP TLS requirement. 
+  # TLS should be enabled for security reasons if using remote endpoint as smarthost.
+  # Disabling tls when using a remote endpoint as smarthost would throw unencrypted connection error.
```
Comment (brian-brazil) [inline content/docs/alerting/configuration.md]: "Note that Go does not support unencrypted connections to remote SMTP endpoints".

Comment (lyveng) [reply content/docs/alerting/configuration.md]: Done

File: content/docs/alerting/configuration.md [hunk 2]
```diff
@@ -78,7 +78,8 @@ global:
   [ smtp_auth_identity: <string> ]
   # SMTP Auth using CRAM-MD5. 
   [ smtp_auth_secret: <secret> ]
-  # The default SMTP TLS requirement.
+  # The default SMTP TLS requirement. 
+  # Note that Go does not support unencrypted connections to remote SMTP endpoints
```
Comment (brian-brazil) [inline content/docs/alerting/configuration.md]: Full stop.

Comment (lyveng) [reply content/docs/alerting/configuration.md]: Done

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**